internal/alb/lb/loadbalancer.go

```diff
@@ -395,11 +395,17 @@ func (controller *defaultController) clusterSubnets(ctx context.Context, scheme
 // subnetIsUsable determines if the subnet shares the same availability zone as a subnet in the
 // existing list. If it does, false is returned as you cannot have albs provisioned to 2 subnets in
 // the same availability zone.
+// Also determine if the subnet has sufficient free IP space available.
 func subnetIsUsable(new *ec2.Subnet, existing []*ec2.Subnet) bool {
 	for _, subnet := range existing {
 		if *new.AvailabilityZone == *subnet.AvailabilityZone {
 			return false
 		}
 	}
+
+	if *new.AvailableIpAddressCount < 8 {
```

Comment:
better to use `aws.Int64Value(new.AvailableIpAddressCount) < 8` to be nil safe.

Reply:
Done.